web/api/v1/api_test.go

```diff
@@ -602,6 +602,37 @@ func TestGetSeries(t *testing.T) {
 			}
 		})
 	}
+
+	mockQuerier := &storage.MockQuerier{
+		SelectMockFunction: func(sortSeries bool, hints *storage.SelectHints, matchers ...*labels.Matcher) storage.SeriesSet {
+			return storage.TestSeriesSetWithSlowAtMethod(storage.MockSeries(
+				[]int64{1, 2, 3},
+				[]float64{1, 2, 3},
+				[]string{"__name__", "test_metric2", "abc", "qwerty", "foo", "baz"},
+			))
+		},
+	}
+
+	api = &API{
+		Queryable: &storage.MockSampleAndChunkQueryable{
+			MockQueryable:      storage.MockQueryable{MockQuerier: mockQuerier},
+			MockChunkQueryable: storage.MockChunkQueryable{},
+		},
+	}
+
+	t.Run("should throw error when context cancelled on SeriesSet iteration", func(t *testing.T) {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		timer := time.AfterFunc(10*time.Millisecond, func() {
```

Comment:
I didn’t know the test would be so dependent on the implementation. I’m afraid that if the `series` body changes, we’ll have no guarantee that the context check you added is the one responsible for the cancellation.

I think we can go without a test for now. Perhaps we can just add a check for the error mapping in TestReturnAPIError. If you’d like, you can create an issue to discuss how we can improve the utils to better test such changes.

Reply:
Agree, this test doesn't look right with timer and at all. Removed test and added the mapping for context.Canceled error case.